app/controllers/webhooks_controller.rb

```diff
@@ -1,21 +1,44 @@
 # frozen_string_literal: true
 module Zoom
   class WebhooksController < ApplicationController
-    skip_before_action :verify_authenticity_token, :redirect_to_login_if_required
+    skip_before_action :verify_authenticity_token,
+                       :redirect_to_login_if_required
     before_action :filter_unhandled,
                   :ensure_webhook_authenticity,
                   :filter_expired_event
 
-    HANDLED_EVENTS = [
-      "webinar.updated",
-      "webinar.started",
-      "webinar.ended",
-      "webinar.participant_joined",
-      "webinar.participant_left"
+    HANDLED_EVENTS = %w[
+      webinar.updated
+      webinar.started
+      webinar.ended
+      webinar.participant_joined
+      webinar.participant_left
+      endpoint.url_validation
     ]
 
     def webinars
-      send(handler_for(webinar_params[:event]))
+      request_params = webinar_params
+      if request_params[:event] == "endpoint.url_validation"
+        secret = SiteSetting.zoom_webhooks_secret_token
+        encrypted_token =
+          OpenSSL::HMAC.hexdigest(
+            OpenSSL::Digest.new("sha256"),
+            secret,
+            JSON.parse(request.body.read, symbolize_names: true)[:payload][
+              :plainToken
+            ]
+          )
+        response.status = 200
+        response.body = {
+          plainToken: request_params[:payload][:plain_token],
+          encryptedToken: encrypted_token
+        }.to_json
+
+        render json: response.body
```

Comment:
I'm pretty sure this can just be 
```ruby
return render json: { 
  plainToken: request_params[:payload][:plain_token],
  encryptedToken: encrypted_token
}, status: 200
```

which will remove a few lines :)